server/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -308,7 +308,7 @@ public static void normalizePoint(double[] lonLat, boolean normLon, boolean norm
         assert lonLat != null && lonLat.length == 2;
 
         normLat = normLat && (lonLat[1] > 90 || lonLat[1] < -90);
-        normLon = normLon && (lonLat[0] > 180 || lonLat[0] < -180);
+        normLon = normLon && (lonLat[0] > 180 || lonLat[0] < -180 || normLat);
```

Comment:
should there be a test reflecting this change in GeoUtilsTests?

Reply:
Good point! I added a couple of edge case tests there.

Reply:
thanks!